gobblin-service/src/main/java/org/apache/gobblin/service/monitoring/KafkaJobStatusMonitor.java

```diff
@@ -426,7 +426,15 @@ public static long getExecutionIdFromTableName(String tableName) {
 
   protected abstract org.apache.gobblin.configuration.State parseJobStatus(GobblinTrackingEvent event);
 
-  public static boolean isThrowableInstanceOf(Throwable exception, List<Class<? extends Exception>> typesList) {
-    return typesList.stream().anyMatch(e -> e.isInstance(exception));
+  public static boolean isThrowableInstanceOf(Throwable exception, List<Class<? extends Exception>> exceptionsList) {
```

Comment:
this changes the semantics.  perhaps that's OK, but a more precise name along w/ some javadoc might help.  e.g. `isThrowableOrCaseInstanceOf` or `isThrowableInstanceOfRecursive`

also, seems better suited to living within a utils class, rather than here in KJSM which was probably just the first place we realized we needed such fucntionality.